sdks/python/apache_beam/runners/worker/sideinputs.py

```diff
@@ -130,7 +163,14 @@ def __iter__(self):
     try:
       while True:
         try:
-          element = self.element_queue.get()
+          if self.element_queue.empty():
+            # We check the current step only when the Queue is empty because we
+            # will likely block waiting (if the queue is not empty, then we'll
+            # be able to get from it instantly). This allows us to limit how
+            # much we check the current step.
+            self.read_counter.check_step()
+          with self.read_counter:
```

Comment:
<!--new_thread; commit:4255543cdc7cfcadd29633f321599d2f591021d8; resolved:0-->
It doesn't look like to me that the changes here are guarded by the experiment flag?  Perhaps we can run a quick profile to verify these new changes don't change performance (particularly here with the "with self.read_counter").